src/tag_index.c

```diff
@@ -59,7 +59,7 @@ char *TagIndex_SepString(char sep, char **s, size_t *toklen) {
 
 static int tokenizeTagString(const char *str, char sep, TagFieldFlags flags, char ***resArray) {
   if (sep == TAG_FIELD_DEFAULT_JSON_SEP) {
-    *resArray = array_append(*resArray, rm_strdup(str));
+    *resArray = array_append(*resArray, strtolower(rm_strdup(str)));
```

Comment:
Shouldn't we also consider `flags & TagField_CaseSensitive` as with none-default separator?
(can also use `tag_strtolower` function?)

Reply:
good call. 
Fixed and tests added.